### DISCUSSION TITLE
Interested, but a few more questions...

### DISCUSSION THREAD

**Discussion by millercentral** [brettpetch/plex-cdn#1] [General]
Really interesting setup, thanks for sharing. I host a plex server in my homelab, but despite a 1gbps symmetrical fiber link, some of my family on the other side of the country appear to have peering issues and are sometimes seeing very low bandwidth to my endpoint. I'm curious if this type of setup could help, but I have some gaps in my understanding...

Assumptions: 
- My plex server is hosted from my homelab, plex.{mydomain}.com port 32400, domain entries managed by cloudflare
- I have an existing LetsEncrypt wildcard cert for {mydomain}.com 
- Assume I want to setup a cdn on the east coast

Questions:
- Do I need to setup a new domain (vs the one I have managed by cloudflare) that is managed by AWS Route 53 service? (seems like it)
- Would "srv1.example.com" in your docs map to my homelab frontdoor IP address?
- The "bouncer ip" is the ip of the VPS in AWS that I am setting up config from your docs on, yes?
- In magic.conf there is a reference to "srv2" -- is that for a hypothetical second plex server or other service? (ie unneeded)?
- What does client configuration look like for the east coast Plex user? Do they have to manually enter the proxy server address?

Thanks in advance for helping me learn!

**Comment by brettpetch** [comment]
Hiya. 

So based on your assumptions, you'll need to change a few things, both domain entries (being managed by cloudflare) is something that you'll want to change to AWS Route 53 for the cheapest reliable GeoIP DNS provider possible. You can bring your own domain.

After changing providers, you'll need to swap over your LE wildcard cert, which you can do (as described in the readme)

srv1.example.com would be mapped to the homelab frontdoor ip, yes.

The "bouncer ip" is the ip of the VPS, which can be from any vendor you'd like. AWS is expensive for VPSes with alot of bandwidth. I'd suggest looking at DigitalOcean, Hetzner, or even OVH depending on your usecase.

magic.conf is a hypothetical second plex server that is being routed through the CDN as well.

> What does client configuration look like for the east coast Plex user? Do they have to manually enter the proxy server address?

The CDN is supposed to resolve to your front door ip AND the VPS ip, dependant on location (which is where aws route53 comes in). You will need the CDN config + nginx on both nodes, then you'll need an additional nginx config (or a secondary mapping on the server actually running plex to the srv1.example.com and srv1.cdn.example.com (or you could do srv1-cdn.example.com) but that's up to you.

Feel free to ask again if there's something that isn't clear.

/b

**Reply by millercentral** [reply]
Okay, super helpful and makes more sense -- Route53 is the DNS management layer (replacement for how I use cloudflare today) with added smarts about Geographic cdn proxying. I believe I now understand that part.

What kind of EC2 server do I need for the nginx proxy (I've experimented with AWS EC2 but am still very much a novice) - it would serve a small handful of users but high bandwidth content. I assume that EC2 instance needs to sit in the regional datacenter we are trying to geo-proxy with Route53, correct?

I'm still a little foggy on how plex app setup works once this is in place. Will it look just the same when connecting to app.plex.tv, but "automagically" route through the proxies for regional users, or do they need to change their setup?

**Reply by brettpetch** [reply]
I wouldn't go near using EC2 or any other kind of AWS compute for sending the data. I simply use it for routing. I'd go to DigitalOcean (droplet), Streamline Servers (Australian VPS provider with pure GTT transport), OVH (VPS), Hetzner Cloud (VPS), RackNerd, or something like that.

https://lowendbox.com/ could be a good resource.

And if they don't have one, ask for a testfile in the location you are looking for a server in!

**Comment by millercentral** [comment]
The goal of this (my understanding anyway) is to provide a proxy that is physically closer to the intended client than your server is, so you would want a VPS in the same region as the client (ideally on a backbone that has better routing to your server than the client does).

The {region} in the vps name – at least for me – is the physical region that vps is located, so for mine in chicago it is vps.us-ch.example.com.

You do also need a “default” proxy for all other regions. I ran this on the same front door as the rest of my homelab services, and I had to switch to a static port-forward port for my plex, rather than a random assigned one (which was something I was overdue to do anyhow since upnp == bad).

My main challenge now is validating how well it works. I see activity in the vps nginx access log when a in region user connects, but the cache directory is empty and I can’t tell if the video streams are proxying through the VPS or still going directly back to server (there is almost no CPU usage on the VPS during playback…)

**Comment by brettpetch** [comment]
Heya,

You might want to ensure that ownership of the directories is correct for image caching to work — the name doesn’t matter as much as the geoip record bouncing the correct region. 

If you enable logging you should be able to see video starts. Ensure that Plex is only advertising the CDN record. Otherwise it’ll be advertising direct connections instead of connecting through the proxy. 

Could be useful to read up on tuning tcp networking on the OS you’ve chosen as well.

**Comment by pmscloudeu** [comment]
Hi, I can't seem to get it to work so must be doing something wrong.

I have a hetzner server in EU and have setup a bouncer VPS in my region.
I am using dockstarter on my hetzner server for docker containers for plex and various other apps, including swag (letsencrypt).
Plex and swag have app variables and I am not sure if they are correct?

![Screenshot 2022-09-08 at 10 22 27](https://user-images.githubusercontent.com/44649340/189074067-fc5d37a5-deca-4691-a30f-d99d6195062e.png)

![Screenshot 2022-09-08 at 10 25 30](https://user-images.githubusercontent.com/44649340/189074202-d4f5cddf-f45f-4bf8-844a-e2c4a89e3530.png)
SWAG_SUBDOMAINS used are i.e. sonarr,plex, etc
SWAG_URL is my domain i.e. example.com

I have followed your guide:

- Route 53 is setup and working
- Letsencrypt installed (hetzner + bouncer vps)
- Nginx configured (hetzner + bouncer vps)
  for the php config, I run **phpv=$(ls -d */ | cut -d/ -f1)**, but do not get/see the below circled:
![Screenshot 2022-09-08 at 10 38 18](https://user-images.githubusercontent.com/44649340/189076714-2ce3a801-ad59-4ea7-960c-2aa0a7e04f6f.png)

- have edited both magic.conf and realip.conf (hetzner + bouncer vps)
- changed advanced network settings in plex to plex.cdn.example.com

But, I am unable to access my plex server on plex.cdn.example.com ?

Any assistance would be appreciated.

Thanks

**Reply by brettpetch** [reply]
Hi there,

The php changes are only required if you want to run a speedtest on the cdn domain.

/b

**Reply by brettpetch** [reply]
To me it looks like you might need to create an intermediate configuration file to bend from port 32400 into port 443, or you could alternatively reference the port 32400 on both the bouncer and the host in your config file, assuming it's available outside of localhost. I'm not quite sure what a dockstarter setup looks like, but i'd be happy to take a look a little later in the evening.

**Reply by pmscloudeu** [reply]
Much appreciated

**Reply by pmscloudeu** [reply]
Any luck? Tx

**Reply by brettpetch** [reply]
Do you have any actual relevant info other than dockstarter screenshots? They don't help. It looks like it uses swag, so you should just be able to add the secondary config and have it talk to Plex directly or through the normal Plex subdomain.

**Reply by pmscloudeu** [reply]
Apologies. Yes, it uses a swag docker container but the only settings I am able to edit are those as shown in the screenshot (which are given during the setup wizard).
I can go into the config folder and edit/add whatever is required for it to work, but am not sure what the secondary config file should look like inside the swag config folder and where it should sit?

Here are a few more screenshots if it helps at all. I have gone into various folders and files:

![Uploading Screenshot 2022-09-13 at 19.39.32.png…]()
<img width="587" alt="Screenshot 2022-09-13 at 19 39 51" src="https://user-images.githubusercontent.com/44649340/189973740-b9cdba0c-16f2-426e-b7d4-710e5ef30be8.png">
<img width="1090" alt="Screenshot 2022-09-13 at 19 40 18" src="https://user-images.githubusercontent.com/44649340/189973773-5bfc3500-7c2a-42f6-9d4a-4d73eed00c86.png">
<img width="1110" alt="Screenshot 2022-09-13 at 19 40 54" src="https://user-images.githubusercontent.com/44649340/189973800-babefe92-641b-48a2-a517-efbf8cb670f9.png">
<img width="1128" alt="Screenshot 2022-09-13 at 19 41 26" src="https://user-images.githubusercontent.com/44649340/189973837-95f1eb99-55d4-44c7-b908-69f8092c4685.png">
<img width="1323" alt="Screenshot 2022-09-13 at 19 43 23" src="https://user-images.githubusercontent.com/44649340/189973863-60623a2c-d83e-4a32-be74-862749934a80.png">
<img width="1354" alt="Screenshot 2022-09-13 at 19 45 24" src="https://user-images.githubusercontent.com/44649340/189973893-268c23df-beff-40f9-8ab9-c877b766c860.png">

Many thanks and apologies for the lack of knowledge on my end

**Reply by brettpetch** [reply]
That helps considerably. This is a drop in for beside your plex.subdomain.conf file. It should be configured similarly.

**Reply by pmscloudeu** [reply]
Thanks for the revert. I unfortunately don't know how/unable to get it to work.

**Reply by brettpetch** [reply]
> Thanks for the revert. I unfortunately don't know how/unable to get it to work.

Join swizzin discord and see if you can find me in #off-topic. I'll try to help you when I have a few minutes to sit down and walk you through it. https://discord.gg/sKjs9UM
